nntrainer/graph/network_graph.h

```diff
@@ -226,6 +298,26 @@ class NetworkGraph {
    */
   void inPlaceOptimize(Manager &manager);
 
+  /**
+   * @brief     Copy the graph
+   * @param[in] from Graph Object to copy
+   * @retval    Graph Object copyed
+   */
+  NetworkGraph &copy(NetworkGraph &from) {
+    if (this != &from) {
+      for (unsigned int i = 0; i < layers.size(); i++)
+        layers[i]->copy(from.layers[i]);
```

Comment:
1. How about copy constructor instead?
2. I wonder if copying layer is enough.

Reply:
This PR just intends to move `layers` and its related code out of model to graph.
More refining is being done in #1016. I will add it here.

> How about copy constructor instead?

Yes, #1016 will do it.

> I wonder if copying layer is enough.

This will depend on the semantics of the layer itself. Currently this is very wrong as this expects `this->layers.size()` is already same as `from.layers.size()`. All this being done in #1016. This PR just moves things out of model to graph without any improvements or major bug fixes.

Reply:
Awesome